### ISSUE TITLE
Scorer: multi-task training pipeline (Phase 1 frozen + Phase 2 fine-tune)

### ISSUE THREAD

**Issue by tartakovsky** [berelevant-ai/slopless#20]
## What

Build the multi-task training pipeline that supports Phase 1 (frozen-encoder, head-only) and Phase 2 (joint fine-tune).

## Why

Each head needs a training loop. A shared pipeline avoids duplicating data loading, label transforms, masked losses, and eval per axis.

## Approach

**Shared components:**

1. **Dataset manifest.** YAML or JSON describing every dataset:
   - source file path
   - axis (specificity / vagueness / informativeness / boilerplate)
   - label type (binary / real-valued / ordinal)
   - split (train / val / test)
   - label transform (normalize to 0..1, etc.)

2. **DataLoader.** Reads the manifest, yields `(sentence, axis_labels)` where `axis_labels` is a dict — entries only present for axes that have labels for this sentence.

3. **Phase 1 loop.** For each axis:
   - Encode all sentences with frozen BGE-small.
   - Train one head (LightGBM / logistic / ridge / MLP) on the pooled vectors → axis label.
   - Report Phase 1 baseline metrics.

4. **Phase 2 loop (gated on Phase 1 results).** Multi-task fine-tune:
   ```
   loss = sum over axes:
     mse_loss(head[axis](encoder(sentence)), label[axis]) if label[axis] is present
     else 0
   ```
   Low LR on encoder (1e-5 to 5e-5), higher LR on heads (1e-3).

5. **Eval harness.** Per-axis metrics per the plan's "Evaluation" section.

## Files (proposed)

- `packages/scorer/training/manifest.yaml`
- `packages/scorer/training/dataloader.py`
- `packages/scorer/training/phase1.py`
- `packages/scorer/training/phase2.py`
- `packages/scorer/training/eval.py`

## Blocked by

- BGE-small encoder integration
- At least one dataset prep complete (Specificity is the most label-rich, start there)

## Done

- Phase 1 trains successfully for at least one axis (specificity) end-to-end.
- Phase 1 metrics logged to a results JSONL.
- Phase 2 stub exists and runs without errors; full fine-tune deferred until Phase 1 results justify it.

## Reference

[Plan section: Training Strategy](https://github.com/agent-quality-controls/slopless/blob/main/.plans/2026-05-13-164606-bge-small-sentence-scorer-prototype.md)
